### DISCUSSION TITLE
Support for MultiPart fields in OpenAPI docs and parsing

### DISCUSSION THREAD

**Discussion by adriangb** [Neoteroi/BlackSheep#208] [General]
Not sure if this is a bug report of feature request.

I would like to deserialize and document (in OpenAPI) a multipart request. The docs [have an example](https://www.neoteroi.dev/blacksheep/requests/#reading-form), but it is not clear if the example is for an `application/x-www-form-urlencoded` request or a `multipart/form-data` request. From my testing (using `requests`, both content types / formats are accepted).

Additionally, the example gives me the following `openapi.yaml`:

```yaml
paths:
    /something:
        post:
            responses: {}
            operationId: create_something
            parameters: []
            requestBody:
                content:
                    application/json:
                        schema:
                            nullable: false
                required: true
```

(side note, the spec requires that there always be at least 1 response documented)

The issue is that the `content` section incorrectly says `application/json`; I think it should be `multipart/form-data` or `application/x-www-form-urlencoded`.

If I replace `SomethingInput` with a Pydantic model, it at lest produces the right schema:

```yaml
paths:
    /something:
        post:
            responses: {}
            operationId: create_something
            parameters: []
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/SomethingInput'
                required: true
```

In this case all that would be missing for basic support is that the content type be fixed.

But for more advanced use cases, it would be nice to get:
1. Support for mixed files / json / other fields
2. Support for parsing out fields, e.g. if a field is a Pydantic model
3. Support for documenting the fields (including their content type, etc.) in the OpenAPI docs

For what it's worth, the way I handled this in my toy web framework project is to make the parameter to `FromForm` be required to be a Pydantic model and each field in the model represents a field in the multipart request. So something like:

```python
class FieldModel(BaseModel):
    foo: str
    bar: int

class MultiPartModel(BaseModel):
    field_one: FromJson[FieldModel]
    field_two: FromFiles  # field_two is a file or files
    field_three: FromForm[...]  # allowing nested multipart/form-data or application/x-www-form-urlencoded


@post("/something")
async def create_something(input: FromForm[MultiPartModel]):
    ...
```

**Comment by RobertoPrevato** [comment]
This is definitely a bug - actually two bugs - thank you for reporting this. Just yesterday evening I published version 1.2.1 with several new features, I will add these two problems to the issues list.

**Reply by RobertoPrevato** [reply]
I just created two issues using the "Create issue from discussion" feature.

**Reply by adriangb** [reply]
glad to hear these are bugs and not unsupported features 😄 

> I just created two issues using the "Create issue from discussion" feature.

I didn't know that was a feature, but I see the button! That's cool.

**Comment by franco-grobler** [comment]
I am having an issue where the FromFiles field is a text field in the OpenAPI spec. Could an example be added to the OpenAPI section of the documentation for handling file uploads?
